mlir/lib/Gradient/Transforms/gradient_lowering.cpp

```diff
@@ -57,13 +58,34 @@ struct GradientLoweringPass : public OperationPass<ModuleOp> {
         registry.insert<linalg::LinalgDialect>();
         registry.insert<index::IndexDialect>();
         registry.insert<tensor::TensorDialect>();
+        registry.insert<scf::SCFDialect>();
+        registry.insert<func::FuncDialect>();
         registry.insert<memref::MemRefDialect>();
         registry.insert<bufferization::BufferizationDialect>();
     }
 
     void runOnOperation() final
     {
         ModuleOp op = getOperation();
+        TypeConverter vectorTypeConverter;
+        vectorTypeConverter.addConversion([](Type type) -> llvm::Optional<Type> {
+            if (MemRefType::isValidElementType(type)) {
+                return type;
+            }
+            return llvm::None;
+        });
+        vectorTypeConverter.addConversion(
+            [](ParameterVectorType type, SmallVectorImpl<Type> &resultTypes) {
+                // Data
+                resultTypes.push_back(MemRefType::get(
+                    {}, MemRefType::get({ShapedType::kDynamic}, type.getElementType())));
+                auto indexMemRef = MemRefType::get({}, IndexType::get(type.getContext()));
+                // Size
+                resultTypes.push_back(indexMemRef);
+                // Capacity
+                resultTypes.push_back(indexMemRef);
+                return success();
```

Comment:
Ok I see now the vector type translates to a memref, a size, and a capacity, so you couldn't just use a cast 👍